src/main/java/io/supertokens/webserver/WebserverAPI.java

```diff
@@ -225,7 +233,26 @@ private String getConnectionUriDomain(HttpServletRequest req) {
     }
 
     protected TenantIdentifier getTenantIdentifier(HttpServletRequest req) {
-        return new TenantIdentifier(this.getConnectionUriDomain(req), this.getAppId(req), this.getTenantId(req));
+        TenantIdentifier tenantIdentifier = new TenantIdentifier(this.getConnectionUriDomain(req), this.getAppId(req), this.getTenantId(req));
+        try {
+            Storage storage = StorageLayer.getStorage(tenantIdentifier, main);
+            tenantIdentifier = tenantIdentifier.withStorage(storage);
+        } catch (TenantOrAppNotFoundException e) {
+            // TODO ignore only for now, this function should throw this exception
+        }
```

Comment:
TODO

Reply:
We can add TODO to the base PR so that this can be merged. This TODO can be done only after all recipes are implemented.